src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Oid.cs

```diff
@@ -87,7 +89,7 @@ public static Oid FromOidValue(string oidValue, OidGroup group)
         {
             get
             {
-                if (_friendlyName == null && _value != null)
+                if (!_hasInitializedFriendlyName && _value != null)
                 {
                     _friendlyName = OidLookup.ToFriendlyName(_value, _group, fallBackToAllGroups: true);
                 }
```

Comment:
One of

```suggestion
                    _hasInitializedFriendlyName = true;
                }
```

or

```suggestion
                    _hasInitializedFriendlyName = _friendlyName != null;
                }
```

Reply:
Doh, good catch. Added a test to cover this.